### PR TITLE
Chapter 3

### PR DESCRIPTION
**Pull request by Rashdan98**
ready

### CODE REVIEW

**Comment by Rashdan98** [comment]
ready

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (ghost) [inline app/assets/stylesheets/static_pages.scss]: please delete this file

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+    def full_title page_title = ""
+      base_title = t  "base_title"
+      page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+    end
+  end
```
Comment (ghost) [inline app/helpers/application_helper.rb]:
```suggestion
   def full_title page_title = ""
     base_title = t  "base_title"
     page_title.empty? ? base_title : "#{page_title} | #{base_title}"
  end
end
```

File: app/helpers/static_pages_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (ghost) [inline app/helpers/static_pages_helper.rb]: please delete this file

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
-    <%= yield %>
+    <%= render "layouts/header" %>
+  <div class="container">
+   <%= yield %>
+   </div>
```
Comment (ghost) [inline app/views/layouts/application.html.erb]: fix indent please

File: test/controllers/static_pages_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (ghost) [inline test/controllers/static_pages_controller_test.rb]: you don't need do test 
please delete this file

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+    def full_title page_title = ""
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: wrong indent
tab 2 space please

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -1,2 +1,6 @@
...
+        base_title = t  "base_title"
+        page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+     end
+   end
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: you are missing end line 
image

File: app/views/static_pages/help.html.erb [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <%= link_to "Rails Tutorial help page", "http://www.railstutorial.org/help" %>
+ To get help on this sample app, see the
+ <a href="http://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em>
+ book</a>.
+ 
+</p>
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: end line please

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <%= link_to "Rails Tutorial help page", "http://www.railstutorial.org/help" %>
+ To get help on this sample app, see the
+ <a href="http://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em>
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: use link_to instead of a tag
https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to

**Comment by ngocvt-0484** [comment]
please change 3 commits to 1 commit

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:1]:
```suggestion
module ApplicationHelper
  def full_title page_title = ""
    base_title = t "base_title"
    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
  end
end
```

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,16 +1,19 @@
...
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+    <%= render "layouts/header" %>
+  <div class="container">
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: wrong indent
```suggestion
    <div class="container">
      <%= yield %>
    </div>
```

File: app/views/static_pages/home.html.erb [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+<h1>Sample App</h1>
+<p>
+  This is the home page for the
+  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+  sample application.
+  
+</p>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: insert new empty line at the end of file

**Review by huongnt-2545 — COMMENTED** [review]
Pull request have 2 commits, you need to combine in 1

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/assets/stylesheets/static_pages.scss (hunk 1 above)
Comment (ngocvt-0484) [reply app/assets/stylesheets/static_pages.scss]: not fix yet

File: app/helpers/application_helper.rb [hunk 13]
```diff
@@ -1,2 +1,6 @@
...
-end
+  def full_title page_title = ""
+    base_title = t "base_title"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+  end
+end
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: please add end line

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,16 +1,24 @@
...
     <%= csp_meta_tag %>
-
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+    <%= render "layouts/header" %>
+    <div class="container">
+<<<<<<< HEAD
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]: fix conflict

File: app/views/static_pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,21 @@
+<%= provide :title, "Help" %>
+<<<<<<< HEAD
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: same
fix conflict please

File: test/controllers/static_pages_controller_test.rb (hunk 5 above)
Comment (ngocvt-0484) [reply test/controllers/static_pages_controller_test.rb]: not fix yet

**Comment by ngocvt-0484** [comment]
image

please fix conflict and comments before ready

File: app/views/static_pages/help.html.erb (hunk 15 above)
Comment (Rashdan98) [reply app/views/static_pages/help.html.erb]: done fix the conflict

File: app/views/layouts/application.html.erb (hunk 14 above)
Comment (Rashdan98) [reply app/views/layouts/application.html.erb]: done

File: app/helpers/application_helper.rb (hunk 13 above)
Comment (Rashdan98) [reply app/helpers/application_helper.rb]: done

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/views/layouts/application.html.erb (hunk 11 above)
Comment (Rashdan98) [reply app/views/layouts/application.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 9 above)
Comment (Rashdan98) [reply app/views/static_pages/help.html.erb]: done

File: app/views/static_pages/home.html.erb [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<%= provide :title, "Home" %>
+<h1>Sample App</h1>
+<p>
+  This is the home page for the
+  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+  sample application.
+<<<<<<< HEAD
+=======
+  
+>>>>>>> chapter_3
```
Comment (ghost) [inline app/views/static_pages/home.html.erb]: remove these line please

File: test/controllers/static_pages_controller_test.rb (hunk 5 above)
Comment (ghost) [reply test/controllers/static_pages_controller_test.rb]: not fix yet

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -1,16 +1,24 @@
...
+      <%= yield %>
+    </div>
+=======
     <%= yield %>
+  </div>
+>>>>>>> chapter_3
```
Comment (ghost) [inline app/views/layouts/application.html.erb]: fix conflict code please

**Review by ghost — COMMENTED** [review]
cmt

File: app/views/static_pages/home.html.erb (hunk 16 above)
Comment (Rashdan98) [reply app/views/static_pages/home.html.erb]: done

File: app/views/layouts/application.html.erb (hunk 17 above)
Comment (Rashdan98) [reply app/views/layouts/application.html.erb]: done

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/controllers/static_pages_controller.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (ghost) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def home; end

  def help; end
```

**Review by ghost — COMMENTED** [review]
Fix all your comment please

File: app/controllers/static_pages_controller.rb (hunk 18 above)
Comment (Rashdan98) [reply app/controllers/static_pages_controller.rb]: alright, I already fix it

**Comment by Rashdan98** [comment]
ready

File: test/controllers/static_pages_controller_test.rb (hunk 5 above)
Comment (ghost) [reply test/controllers/static_pages_controller_test.rb]: not fix yet

File: app/views/static_pages/help.html.erb [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
+help page</a>.
+To get help on this sample app, see the
+<a href="http://www.railstutorial.org/book"><em>Ruby on Rails
+Tutorial</em>
+book</a>.
```
Comment (ghost) [inline app/views/static_pages/help.html.erb]: indent please

File: app/views/layouts/application.html.erb [hunk 20]
```diff
@@ -1,16 +1,22 @@
...
+
+    <%= render "layouts/header" %>
+
+    <div class="container">
+      <%= yield %>
+    </div>
+
   </body>
-</html>
+</html>
```
Comment (ghost) [inline app/views/layouts/application.html.erb]: fix endline please

File: app/controllers/static_pages_controller.rb [hunk 21]
```diff
@@ -0,0 +1,4 @@
+class StaticPagesController < ApplicationController
+  def home; end
+
+  def help; end
```
Comment (ghost) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def help; end
end
```

**Review by ghost — COMMENTED** [review]
cmt

File: app/views/static_pages/help.html.erb (hunk 19 above)
Comment (Rashdan98) [reply app/views/static_pages/help.html.erb]: alright, done

File: app/views/layouts/application.html.erb (hunk 20 above)
Comment (Rashdan98) [reply app/views/layouts/application.html.erb]: at line 16, do i need to delete the line? is it correct i do this?

File: app/controllers/static_pages_controller.rb (hunk 21 above)
Comment (Rashdan98) [reply app/controllers/static_pages_controller.rb]: done

**Comment by Rashdan98** [comment]
ready

File: app/controllers/static_pages_controller.rb [hunk 22]
```diff
@@ -0,0 +1,5 @@
...
+end
```
Comment (ghost) [inline app/controllers/static_pages_controller.rb]: please add a blank line at end of the file to  fix this

**Review by ghost — COMMENTED** [review]
cmt

File: app/controllers/static_pages_controller.rb (hunk 22 above)
Comment (Rashdan98) [reply app/controllers/static_pages_controller.rb]: alright, done

**Comment by Rashdan98** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**